imap_processing/mag/l0/decom_mag.py

```diff
@@ -53,9 +62,89 @@ def decom_packets(packet_file_path: str) -> list[MagL0]:
                     ]
                     data_list.append(MagL0(CcsdsData(packet.header), *values))
         except ReadError as e:
-            logger.error(
+            logger.warning(
                 f"Found error: {e}\n This may mean reaching the end of an "
                 f"incomplete packet."
             )
 
         return data_list
+
+
+def export_to_xarray(l0_data: list[MagL0]):
+    """Generate xarray files for "raw" MAG CDF files from MagL0 data.
+
+    Mag outputs "RAW" CDF files just after decomming. These have the immediately
+    post-decom data, with raw binary data for the vectors instead of vector values.
+
+    Parameters
+    ----------
+    l0_data: list[MagL0]
+        A list of MagL0 datapoints
+    """
+    # TODO split by mago and magi using primary sensor
+    # TODO split by norm and burst
+    norm_data = defaultdict(
+        list
+    )  # dict.fromkeys(dataclasses.asdict(l1a_data[0]).keys(), [])
+    burst_data = norm_data.copy()
+
+    for datapoint in l0_data:
+        if datapoint.ccsds_header.PKT_APID == Mode.NORM:
+            for key, value in dataclasses.asdict(datapoint).items():
+                if key != "ccsds_header":
+                    norm_data[key].append(value)
+        if datapoint.ccsds_header.PKT_APID == Mode.BURST:
+            burst_data["SHCOARSE"].append(datapoint.SHCOARSE)
+            burst_data["raw_vectors"].append(datapoint.VECTORS)
+
+    # Used in L1A vectors
+    direction = xr.DataArray(
+        np.arange(194),
+        name="Direction",
+        dims=["Direction"],
+        attrs=mag_cdf_attrs.direction_attrs.output(),
+    )
+
+    # print(direction)
+    norm_epoch_time = xr.DataArray(
+        norm_data["SHCOARSE"],
+        name="Epoch",
+        dims=["Epoch"],
+        attrs=ConstantCoordinates.EPOCH,
+    )
+
+    # TODO: raw vectors units
+    norm_raw_vectors = xr.DataArray(
+        norm_data["VECTORS"],
+        name="Raw_Vectors",
+        dims=["Epoch", "Direction"],
+        attrs=mag_cdf_attrs.mag_vector_attrs.output(),
+    )
+
+    # TODO add norm to attrs somehow
+    norm_dataset = xr.Dataset(
+        coords={"Epoch": norm_epoch_time, "Direction": direction},
+        attrs=mag_cdf_attrs.mag_l1a_attrs.output(),
+    )
+
+    norm_dataset["RAW_VECTORS"] = norm_raw_vectors
+
+    # TODO: retrieve the doc for the CDF description
+    print(getattr(MagL0, "__doc__", {}))
```

Comment:
Do you want this print there? If so, I'd add some extra context, otherwise if it is debugging I'd remove it and leave the TODO.

Reply:
Ah yeah that was left in for future work, but shouldn't be merged. I'll remove it and add it in the TODO or my notes